addons/account/tests/test_account_move_out_invoice.py

```diff
@@ -4263,3 +4263,28 @@ def test_out_invoice_bank_partner(self):
             bank_2,
             invoice_new.partner_bank_id
         )
+
+    def test_narration_preserved_when_use_invoice_terms_disabled(self):
+        """ Ensure narration is preserved when partner changes and invoice terms are disabled. """
+        self.env['ir.config_parameter'].sudo().set_param('account.use_invoice_terms', False)
+        invoice = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'partner_id': self.partner_a.id,
+            'journal_id': self.company_data['default_journal_sale'].id,
+            'invoice_date': fields.Date.today(),
+            'invoice_line_ids': [(0, 0, {
+                'product_id': self.product_a.id,
+                'price_unit': 100,
+                'quantity': 1,
+                'account_id': self.product_a.property_account_income_id.id,
+            })],
+            'narration': 'Manually written terms by user',
+        })
```

Comment:
Nitpick - but it saves 10 lines :)
```suggestion
        invoice = self.invoice.copy({
            'narration': 'Manually written terms by user',
        })
```